commands/run.sh

```diff
@@ -348,9 +348,9 @@ fi
 
 if [[ "${BUILDKITE_PLUGIN_DOCKER_ALWAYS_PULL:-false}" =~ ^(true|on|1)$ ]] ; then
   echo "--- :docker: Pulling ${image}"
-  if ! retry "${BUILDKITE_PLUGIN_DOCKER_PULL_RETRIES:-3}" \
-       docker pull "${image}" ; then
-    retry_exit_status="$?"
+  retry "${BUILDKITE_PLUGIN_DOCKER_PULL_RETRIES:-3}" docker pull "${image}"
```

Comment:
due to the `set -e` at the very top, if the pull fails it will abort the whole script before printing out the error message because you removed the `if`

Reply:
Thanks for the suggestions!

Reply:
Added tests for both success and failure conditions on retries of the `docker pull` command.